Neos.Media/Classes/Domain/Model/AssetSource/AssetSourceInterface.php

```diff
@@ -36,6 +36,22 @@ public function getIdentifier(): string;
      */
     public function getLabel(): string;
 
+    /**
+     * Returns the resource path to the icon of the asset source
```

Comment:
But then we have to check via `methodExists` if the AssetSource has those properties instead of checking for the interface. Is that the expected behaviour?

Reply:
Given, that we do not want to have a `DescriptiveAssetSourceInterface` forever - and I think we shouldn't, I would prefer that in this case. To just force the methods with next major makes upgrades much easier. You don't have to add the `DescriptiveAssetSourceInterface` now to show an Icon and then remove it again with next major. With next major you can remove either the interface or the method checks. 

That was my thoughts behind it, but I am also open to revert the last commit if you object.

Reply:
Given this is just temporary I don't really mind the one or the other way to go. It is true that having to add an interface now and remove it again later is more cumbersome for 3rd parties, and other than some code pureness and a tiny little bit of help by the IDE to add the methods I see no real other benefit of the (temporary) interface.